src/lib.rs

```diff
@@ -101,63 +95,170 @@ impl Router {
+                .map_err(|e| {
+                    log::warn!(
+                        "Error while trying to create temporary directory for module cache: {}",
+                        e
+                    );
+                    e
+                })
+                .map(|td| td.into_path())
+                .unwrap_or_default(),
+            base_log_dir: tempfile::tempdir()
+                .map_err(|e| {
+                    log::warn!(
+                        "Error while trying to create temporary directory for logging: {}",
+                        e
+                    );
+                    e
+                })
+                .map(|td| td.into_path())
+                .unwrap_or_default(),
+            default_host: String::from("localhost:3000"),
+            global_env_vars: HashMap::new(),
+        }
     }
+}
 
-    let data = std::fs::read_to_string(filename)?;
-    let mut modules: ModuleConfig = toml::from_str(data.as_str())?;
+impl RouterBuilder {
+    /// Sets a location for the wasmtime config cache
+    pub fn cache_config_path(mut self, cache_config_path: impl AsRef<Path>) -> Self {
+        self.cache_config_path = cache_config_path.as_ref().to_owned();
+        self
+    }
 
-    modules
-        .build_registry(cache_config_path, module_cache_dir, base_log_dir)
-        .await?;
+    /// Sets a location for caching downloaded Wasm modules
+    pub fn module_cache_dir(mut self, module_cache_dir: impl AsRef<Path>) -> Self {
+        self.module_cache_dir = module_cache_dir.as_ref().to_owned();
+        self
+    }
 
-    Ok(modules)
-}
+    /// Sets the base log directory which is used as a location for storing module logs in unique
+    /// subdirectories
+    pub fn base_log_dir(mut self, base_log_dir: impl AsRef<Path>) -> Self {
+        self.base_log_dir = base_log_dir.as_ref().to_owned();
+        self
+    }
 
-pub async fn load_bindle(
-    name: &str,
-    bindle_server: &str,
-    cache_config_path: String,
-    module_cache_dir: PathBuf,
-    base_log_dir: impl AsRef<Path>,
-) -> Result<ModuleConfig, anyhow::Error> {
-    log::trace!("Loading bindle {}", name);
-    let cache_dir = module_cache_dir.join("_ASSETS");
-    let mut mods = runtime::bindle::bindle_to_modules(name, bindle_server, cache_dir)
-        .await
-        .map_err(|e| anyhow::anyhow!("Failed to turn Bindle into module configuration: {}", e))?;
-
-    mods.build_registry(cache_config_path, module_cache_dir, base_log_dir)
+    /// Sets the default host to use for virtual hosting. If this is already set in a
+    /// `ModulesConfig`, this will be ignored when the router is built
+    pub fn default_host(mut self, host: &str) -> Self {
+        self.default_host = host.to_owned();
+        self
+    }
+
+    pub fn global_env_vars(mut self, vars: HashMap<String, String>) -> Self {
+        self.global_env_vars = vars;
+        self
+    }
+
+    /// Build the router, loading the config from a toml file at the given path
```

Comment:
Super nitpicky (but also a result of my having just read some really frustrating Go code): Maybe `build_from_modules_toml` and `build_from_bindle`? My first read of this was like, "wait... this isn't building a bindle!"

Reply:
I added this just to make sure it was clear